Integrations/python/deephaven/learn/__init__.py

```diff
@@ -14,6 +14,7 @@
 _Output_ = None
 _Computer_ = None
 _Scatterer_ = None
+gatherer = None
```

Comment:
I think what you have here functions, but I don't think it is what we want to deliver.  Let me try to explain some of the vision.

1) We want the UX to be very pythonic.  By exposing the Java `Gatherer`, we have Java naming conventions and other mess that Jianfeng is trying hard to get rid of.

2) This only supports `float64` tensors.  We need to support the spectrum of tensor types.

3) From the Python side, there is way too mutch futzing around to do what you want.  I think we need a module or modules that contain key helper functions.  In this case, maybe the module is `gather` or `input`, and it would contain various utilities for dealing with input data.  

I would guess that these helper functions look something like the following.
```
def buffer_2d(idx, cols, dtype):
    ...

def numpy_2d(idx, cols, dtype):
    buffer =buffer_2d(idx, cols, dtype)
    tensor = np.frombuffer(buffer, dtype = float)
    tensor.shape = (idx.getSize(), len(cols))
    return tensor
```

Then the user code is as simple as:
```
from deephaven.learn.gather import numpy_2d

my_np_gatherer = lambda idx, cols: numpy_2d(idx, cols, np.int32)
my_tf_gatherer = labda idx, cols: tf.convert_to_tensor(numpy_2d(idx, cols, np.float64))
```

Many different `dtype` specifications should be supported.  numpy types?  DH types?  etc. 

4) The java method currently returns a method shaped like the table.  Should there be an argument to return the transpose for efficiency or convenience?  Does `numpy.transpose` or the equivalent TensorFlow or PyTorch command result in extra memory allocation or slowness from indirection?

Reply:
The latest push has everything discussed here except for point 4.  I'm not sure what exactly how we would want to implement a transpose or other operation.  Let's discuss it soon

Reply:
Transpose is very easy.  It would be something like:

```
                final int ij = transpose ? nRows * j + i: nCols * i + j;
```
And then you would also need to set the transposed shape.

Reply:
I think this `gatherer` line is not needed.